javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -204,6 +209,10 @@ public AutoBuild() {
     this.typeScriptMode =
         getEnumFromEnvVar("LGTM_INDEX_TYPESCRIPT", TypeScriptMode.class, TypeScriptMode.FULL);
     this.defaultEncoding = getEnvVar("LGTM_INDEX_DEFAULT_ENCODING");
+    this.installDependencies = Boolean.valueOf(getEnvVar("LGTM_TYPESCRIPT_INSTALL_DEPS"));
```

Comment:
Would it make sense to call the environment variable `LGTM_INDEX_INSTALL_TYPESCRIPT_DEPS` or something else that starts with `LGTM_INDEX_`? Then we could eventually expose it through `lgtm.yml`.

Reply:
It wasn't my intention to expose this through `lgtm.yml`, although I suppose it might be nice to have a way to opt out once we set it in the worker config.

Reply:
Can you explain a bit more why you think this is something we'd want to set in the worker config? I would have expected it to be a per-project setting.

Reply:
We want to turn it on for all projects on LGTM.com, but for LGTM-e it should remain opt-in for now (there's some setup involved there, like installing yarn and getting internet access through a proxy). The worker config settings seemed like the best way to distinguish between LGTM and LGTM-e here.

The setting should not be overridden by a lgtm.yml file with unrelated changes, so setting a global project config therefore won't work.

For the initial canary launch I'd just do `export LGTM_TYPESCRIPT_INSTALL_DEPS=true` in `before_index`.

But again, I'm fine with having a per-project setting that can override the global setting, but maybe it can wait until after the canary launch?

Reply:
I'm OK with deciding this later. If we rename to `LGTM_INDEX_*` we keep all options open (but since it's an undocumented variable we can, of course, rename later as well).